DateTime.mqh

```diff
@@ -113,9 +113,14 @@ class DateTime {
       _result |= DATETIME_SECOND;
     }
 
-    if (dt_curr.GetValue(DATETIME_DAY | DATETIME_WEEK) != dt_last.GetValue(DATETIME_DAY | DATETIME_WEEK)) {
-      // New week started.
-      _result |= DATETIME_WEEK;
+    if (dt_curr.GetValue(DATETIME_DAY | DATETIME_WEEK) == 0) {
```

Comment:
I think many forex brokers don't have ticks on Sundays, maybe it would not work?

Reply:
Forex market opens on Sunday at 10:00 pm GMT.
http://www.forexmarkethours.com/GMT_hours/02/

Reply:
My brokers are using timezone +3:00, and the candles start from 2022-05-23 00:00:00 this week.
Not sure if the code is ok

Reply:
DateTime in general doesn't deal with timezones, the datetime is passed into constructor, so it expects the time in GMT. So for the time being the code initializing the class should pass the date in GMT timezone. The next release is closing soon, so I'm going to merge that to include this fix into the next release, and we can further improve the logic after more testing.